### ISSUE TITLE
Changes to fpy.py

### ISSUE THREAD

**Issue by bclyons12** [nferraro/fusion-io#8]
@andreas-kleiner I'm thinking of making a few changes to how `fpy.py` works. Let me know your thoughts.

1. Store the absolute path as `sim_data.filename`, that way it can be accessed if the filename is required. This would mean that we don't have to pass around `sim_data` and `filename` separately and it would ensure consistency.

2. Initialize `sim_data.hint` to `None`, then instead of calling `sim_data.hint` directly, we do `sim_data.get_hint()`. That function will return `sim_data.hint`, but if it's `None`, it will call `fio_py.allocate_hint()`. By moving this outside of `__init__` we could initialize the `sim_data` object and use it for anything that doesn't require a `hint` without waiting for the mesh connectivity to be computed. That would only get done when it's used.

**Comment by bclyons12** [comment]
Note that 2. would allow us to do `time_trace_fast` without having to access the HDF5 file ourselves.

**Comment by andreas-kleiner** [comment]
@bclyons12 Sounds both good to me. The functions to read the HDF5 files should still be kept though, because they can be useful to diagnose errors and inspect the HDF files.

**Comment by bclyons12** [comment]
Closing with 785bd6de1451b264ebfb0b49842efd693f15137e.

Note that I misunderstood the purpose of `sim_data.hint`. The mesh connectivity is actually calculated when a field first gets accessed. Thus, in order not to compute mesh connectivity during `sim_data.__init__()`, I removed the `sim_data.time` attribute, which was not used as far as I could tell. Let me know if you have any objection @andreas-kleiner
